cmd/root.go

```diff
@@ -54,6 +54,7 @@ func Execute(ctx context.Context, version string) {
 	rootCmd.Flags().BoolVar(&input.useGitIgnore, "use-gitignore", true, "Controls whether paths specified in .gitignore should be copied into container")
 	rootCmd.Flags().StringArrayVarP(&input.containerCapAdd, "container-cap-add", "", []string{}, "kernel capabilities to add to the workflow containers (e.g. --container-cap-add SYS_PTRACE)")
 	rootCmd.Flags().StringArrayVarP(&input.containerCapDrop, "container-cap-drop", "", []string{}, "kernel capabilities to remove from the workflow containers (e.g. --container-cap-drop SYS_PTRACE)")
+	rootCmd.Flags().StringVarP(&input.containerUser, "container-user", "", "runner", "UID or username which will be used for running act containers")
```

Comment:
This will fail on image that doesn't have `runner`

Reply:
Should be `0`, otherwise any other value has chance to break

Reply:
I tested with `ghcr.io/catthehacker/ubuntu:full-latest` because that has everything I need for my workflows.  But this has some coincidental features that made everything work.  My local UID is 1000, and ubuntu:full-latest has a user with that UID (runneradmin).  This allows workflows using `sudo` to succeed.

When my local UID doesn't exist in a container, such as `ubuntu:act-latest`, I get errors like `sudo: you do not exist in the passwd database`.

It seems to me that in order to make this work in the general case and be compatible with GitHub's environment, we would need to:

1. Create a `runner` user (the name is arbitrary) during container setup if it doesn't exist
2. Add `runner` to sudoers if it's not already there
3. Change the UID of `runner` to match `--container-user`
4. Change ownership of the working directory and runner home directory to match that UID

What are your thoughts on that?

I'm also starting to wonder what the advantage of the `--container-user` option is.  For me, I only ever want to use it in combination with `--bind`, set to my local UID, so that I can get the output of a job with the correct local ownership.  Is there another major use-case for it?

If not, perhaps we could drop it and always use the local user UID.  In my opinion, this would make `act` work for workflows where it currently doesn't, without the need for users to start using new command-line options.

Reply:
> I'm also starting to wonder what the advantage of the `--container-user` option is.

The advantage of not breaking program completely.